core/src/main/java/com/datastrato/gravitino/storage/kv/KvGarbageCollector.java

```diff
@@ -232,6 +253,71 @@ private void collectAndRemoveOldVersionData() throws IOException {
         kvBackend.delete(kv.getKey());
       }
     }
+
+    commitIdHasBeenCollected = kvs.isEmpty() ? startKey : kvs.get(0).getKey();
+    kvBackend.put(LAST_COLLECT_COMMIT_ID_KEY, commitIdHasBeenCollected, true);
+  }
+
+  /**
+   * Remove all versions of the key.
+   *
+   * @param rawKey raw key, it contains the transaction id.
+   * @param key key, it's the real key and does not contain the transaction id
+   * @param includeStart whether include the start key.
+   * @throws IOException if an I/O exception occurs during deletion.
+   */
+  private void removeAllVersionsOfKey(byte[] rawKey, byte[] key, boolean includeStart)
+      throws IOException {
+    List<Pair<byte[], byte[]>> kvs =
+        kvBackend.scan(
+            new KvRange.KvRangeBuilder()
+                .start(rawKey)
+                .end(generateKey(key, 1))
+                .startInclusive(includeStart)
+                .endInclusive(false)
+                .build());
+
+    for (Pair<byte[], byte[]> kv : kvs) {
+      // Delete real data.
+      kvBackend.delete(kv.getKey());
+
+      LogHelper logHelper = decodeKey(kv.getKey());
+      LOG.info(
+          "Physically delete key that has marked deleted: name identifier: '{}', entity type: '{}', createTime: '{}({})', key: '{}'",
```

Comment:
This line might be longer than 100 characters?

Reply:
It seems that spotless does not work for it, I 'll make the necessary changes.